### PR TITLE
auto load examples from @angular/components-examples

### PR DESCRIPTION
**Pull request by annieyw**

### CODE REVIEW

File: src/app/shared/documentation-items/documentation-items.ts [hunk 1]
```diff
@@ -66,111 +75,78 @@ const DOCS: {[key: string]: DocCategory[]} = {
...
+          },
         },
         {
           id: 'radio',
           name: 'Radio button',
           summary: 'Allows the user to select one option from a group.',
-          examples: ['radio-ng-model'],
+          exampleSpecs: {
+            prefix: 'radio-',
+            exclude: [],
```
Comment (atscott) [inline src/app/shared/documentation-items/documentation-items.ts:122]: If all the `exclude` arrays are empty, should they just be removed?

File: src/app/shared/documentation-items/documentation-items.ts [hunk 2]
```diff
@@ -361,37 +340,35 @@ const DOCS: {[key: string]: DocCategory[]} = {
           id: 'bottom-sheet',
           name: 'Bottom Sheet',
           summary: 'A large interactive panel primarily for mobile devices.',
-          examples: ['bottom-sheet-overview']
+          exampleSpecs: {
+            prefix: 'bottom-sheet',
```
Comment (atscott) [inline src/app/shared/documentation-items/documentation-items.ts:344]: Curious why some prefixes end with a `-` but others don't. Is this intended?

File: src/app/shared/documentation-items/documentation-items.ts [hunk 3]
```diff
@@ -449,88 +415,102 @@ const DOCS: {[key: string]: DocCategory[]} = {
           id: 'a11y',
           name: 'Accessibility',
           summary: 'Utilities for screen readers, focus and more.',
-          examples: []
+          exampleSpecs: {
+            prefix: 'focus-monitor',
+            exclude: [],
+          },
+          // examples: []
```
Comment (atscott) [inline src/app/shared/documentation-items/documentation-items.ts:422]: nit: remove

File: src/app/shared/documentation-items/documentation-items.ts (hunk 2 above)
Comment (annieyw) [reply src/app/shared/documentation-items/documentation-items.ts:344]: Oops it was a mistake. Now they all end with `-`

File: src/app/shared/documentation-items/documentation-items.ts [hunk 4]
```diff
@@ -601,12 +570,20 @@ const DOCS: {[key: string]: DocCategory[]} = {
 for (const category of DOCS[COMPONENTS]) {
   for (const doc of category.items) {
     doc.packageName = 'material';
+    doc.examples =
+      exampleNames
+        .filter(key => key.match(doc.exampleSpecs.prefix) &&
```
Comment (mmalerba) [inline src/app/shared/documentation-items/documentation-items.ts:575]: probably should be `.startsWith` to make sure we really are treating it as a prefix

Comment (annieyw) [reply src/app/shared/documentation-items/documentation-items.ts]: `.startsWith` not support on IE :( 
added `^` to regex to match the start.

File: src/app/shared/documentation-items/documentation-items.ts [hunk 5]
```diff
@@ -601,12 +570,20 @@ const DOCS: {[key: string]: DocCategory[]} = {
 for (const category of DOCS[COMPONENTS]) {
   for (const doc of category.items) {
     doc.packageName = 'material';
+    doc.examples =
+      exampleNames
+        .filter(key => key.match(/^${doc.exampleSpecs.prefix}/) &&
```
Comment (mmalerba) [inline src/app/shared/documentation-items/documentation-items.ts:575]: I don't think this syntax works in regexes, you'll need to do ``RegExp(`^${...}`)``

**Pull request merged**